### PR TITLE
Theedge 2136 update all devices from group

### PR DESCRIPTION
**Pull request by acosferreira**
# Description

Include a new route to be able to update all devices from a group.

Fixes # (THEEDGE-2136)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [X] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `go fmt ./...` to check that my code is properly formatted
- [X] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
35ab3765-28c9-44b3-b1cb-1970a715249c
5aedfff9-e868-4176-ab96-2dde2bc71709

**Comment by InsightsDroid** [comment]
Test results are available in [Ibutsu](https://url.corp.redhat.com/ibutsu-runs). The test run IDs are:
b145596a-9c24-47c3-922f-24b63e99d661
cbb632b1-46e5-48a8-be39-25c9ea1a90a3

File: pkg/services/updates.go [hunk 1]
```diff
@@ -550,3 +555,164 @@ func (s *UpdateService) ValidateUpdateSelection(account string, imageIds []uint)
...
+			CommitID: devicesUpdate.CommitID,
+			Status:   models.UpdateStatusCreated,
+		}
+
+		// Get the models.Commit from the Commit ID passed in via JSON
+		update.Commit = commit
+
+		notify, errNotify := s.SendDeviceNotification(&update)
+		if errNotify != nil {
+			s.log.WithField("message", errNotify.Error()).Error("Error to send notification")
```
Comment (thom-at-redhat) [inline pkg/services/updates.go:595]: Error sending device notification?  Account? and should the message below be combined with this one?

File: pkg/services/updates.go [hunk 2]
```diff
@@ -550,3 +555,164 @@ func (s *UpdateService) ValidateUpdateSelection(account string, imageIds []uint)
...
+			var updateDevice *models.Device
+			dbDevice := db.DB.Where("uuid = ?", device.ID).First(&updateDevice)
+			if dbDevice.Error != nil {
+				s.log.WithField("error", result.Error.Error()).Error("Error finding device")
+				return nil, new(DeviceNotFoundError)
+			}
+
+			if err != nil {
+				if !(err.Error() == "Device was not found") {
+					s.log.WithField("error", err.Error()).Error("Device was not found in our database")
```
Comment (thom-at-redhat) [inline pkg/services/updates.go:635]: Device was not found in this account?

File: pkg/services/updates.go [hunk 3]
```diff
@@ -550,3 +555,164 @@ func (s *UpdateService) ValidateUpdateSelection(account string, imageIds []uint)
...
+					if result.Error != nil {
+						if result.Error.Error() != "record not found" {
+							s.log.WithField("error", err.Error()).Error("Error returning old commit for this ostree checksum")
+							err := errors.NewBadRequest(err.Error())
+
+							return nil, err
+						}
+					}
+					if result.RowsAffected == 0 {
+						s.log.Debug("No old commits found")
```
Comment (thom-at-redhat) [inline pkg/services/updates.go:694]: Do we want to add the search criteria to the log message as fields?

File: pkg/services/updates.go [hunk 4]
```diff
@@ -550,3 +555,164 @@ func (s *UpdateService) ValidateUpdateSelection(account string, imageIds []uint)
...
+						oldCommits = append(oldCommits, oldCommit)
+					}
+				}
+			}
+
+			if toUpdate {
+				//Should not create a transaction to device already updated
+				update.OldCommits = oldCommits
+				if err := db.DB.Save(&update).Error; err != nil {
+					err := errors.NewBadRequest(err.Error())
```
Comment (thom-at-redhat) [inline pkg/services/updates.go:705]: Should err be reused  in this way?

**Review by thom-at-redhat — CHANGES_REQUESTED** [review]

File: pkg/services/updates.go (hunk 3 above)
Comment (acosferreira) [reply pkg/services/updates.go:694]: I dont think so, is all related to the current device uuid

**Review by thom-at-redhat — APPROVED** [review]

**Comment by thom-at-redhat** [comment]
@mergifyio unqueue

**Comment by thom-at-redhat** [comment]
@mergifyio requeue

**Pull request merged**